tests/unit/onyxMultiMergeWebStorageTest.js

```diff
@@ -0,0 +1,70 @@
+import OnyxCache from '../../lib/OnyxCache';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+import localforageMock from '../../__mocks__/localforage';
+
+const ONYX_KEYS = {
+    COLLECTION: {
+        TEST_KEY: 'test_',
+    },
+};
+
+const initialTestObject = {a: 'a'};
+const initialData = {
+    test_1: initialTestObject,
+    test_2: initialTestObject,
+    test_3: initialTestObject,
+};
+
+localforageMock.setInitialMockData(initialData);
+
+describe('Onyx.mergeCollection()', () => {
+    let Onyx;
+    let Storage;
+    beforeAll(() => {
+        jest.mock('../../lib/storage');
+        Onyx = require('../../index').default;
+        Storage = require('../../lib/storage').default;
+        jest.useRealTimers();
+    });
+
+    it('mergeCollection', () => {
+        expect(Storage.name).toBe('WebStorage');
+
+        Onyx.init({
+            keys: ONYX_KEYS,
+            registerStorageEventListener: () => {},
+            initialKeyStates: {},
+        });
+
+        const additionalDataOne = {b: 'b', c: 'c'};
+        Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {
+            test_1: additionalDataOne,
+            test_2: additionalDataOne,
+            test_3: additionalDataOne,
+        });
+
+        const additionalDataTwo = {d: 'd'};
+        Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {
+            test_1: additionalDataTwo,
+            test_2: additionalDataTwo,
+            test_3: additionalDataTwo,
+        });
+
+        return waitForPromisesToResolve()
+            .then(() => {
+                const finalObject = {
+                    a: 'a', b: 'b', c: 'c', d: 'd',
+                };
+
+                // Expect that our new data has merged with the existing data in the cache
+                expect(OnyxCache.getValue('test_1')).toEqual(finalObject);
+                expect(OnyxCache.getValue('test_2')).toEqual(finalObject);
+                expect(OnyxCache.getValue('test_3')).toEqual(finalObject);
+
+                // Check the storage to make sure our data has been saved
+                expect(localforageMock.storageMap.test_1).toEqual(finalObject);
+                expect(localforageMock.storageMap.test_2).toEqual(finalObject);
+                expect(localforageMock.storageMap.test_3).toEqual(finalObject);
```

Comment:
What I'm saying is changes made around mocking are not necessary here, because which mock is used shouldn't matter
We can assert that `Storage.multiSet` and/or `Storage.multiMerge` are getting called correctly

The changes made to `__mocks__/localforage.js` can help us test `LocalForage.multiSet` and `LocalForage.multiMerge` are working correctly with the queues here: [LocalForageProviderTest](https://github.com/Expensify/react-native-onyx/blob/032ff54b4374c93ee12676792a6e029c61ea1e06/tests/unit/storage/providers/LocalForageProviderTest.js#L1-L6)

Reply:
I am probably not understanding the message. The test is helping me to verify the changes by using a mock that works similar to storage. Why is it not necessary? The previous tests did not test this scenario and things were broken. So I'm adding a case to cover the thing that was breaking. What should we do instead?

Reply:
By testing what's stored inside `storageMap` of the mock we're asserting that `WebStorage` works correctly when `Storage.multiMerge` is called within `mergeCollection` 

We can have a test where we import `LocalForage` and invoke `multiMerge` on it directly

Reply:
I like this test because it more accurately represents the code flow that went wrong here. It's a combination of Onyx changes and storage provider changes so I think it is reasonable to test them both. But I am possibly being lazy 😄

Reply:
Yeah, that makes sense it's just that we're not doing it anywhere else - using specific storage provider and mock

So it's not clear when or why tests need to use a specific storage provider, e.g. why this tests does but others don't?

What we see changed in Onyx.mergeCollection is a change that should work regardless of the storage provider
We can write 2x tests with the other provider as well, but then it results in more maintenance outweighing the value of the tests

I think we can achieve the same test by:
1. Mock `Storage.getItem`
2. Verity `Storage.getItem` is being called when a given key is not available in cache
3. Verify returned mock data is used in cache - e.g. cache matches expected final value

To me this gives us more information regarding what we expect to happen 
_(First time here I didn't even realize we we were testing the case where we don't have data in cache but have it on disk)_